app/uk/gov/hmrc/apisubscriptionfields/controller/ApiSubscriptionFieldsController.scala

```diff
@@ -17,25 +17,65 @@
 package uk.gov.hmrc.apisubscriptionfields.controller
 
 import java.util.UUID
+import javax.inject.{Inject, Singleton}
 
-import play.api.mvc.Action
-import uk.gov.hmrc.play.microservice.controller.BaseController
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc._
+import uk.gov.hmrc.apisubscriptionfields.model.ErrorCode._
+import uk.gov.hmrc.apisubscriptionfields.model._
+import uk.gov.hmrc.apisubscriptionfields.service.SubscriptionFieldsService
 
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-trait ApiSubscriptionFieldsController extends BaseController {
-
-	def id() = Action.async { implicit request =>
-    /*
-     TODO:
-       parse `ApiSubscriptionRequest`
-       - [application_id]_[api_context]_[api_context] is the id of the mongo record, but decide good separators
-       - generate UUID (the header to be passed in the incoming requests)
-       - save record to mongo
-       */
-    val subscriptionFieldsId = UUID.randomUUID()
-    Future.successful(Created(s"$subscriptionFieldsId"))
+@Singleton
+class ApiSubscriptionFieldsController @Inject() (service: SubscriptionFieldsService) extends CommonController {
+
+  import JsonFormatters._
+
+  private val NOT_FOUND_RESPONSE = NotFound(JsErrorResponse(SUBSCRIPTION_FIELDS_ID_NOT_FOUND, "Subscription Fields were not found"))
+
+  def id(rawAppId: UUID, rawApiContext: String, rawApiVersion: String): Action[AnyContent] = Action.async { implicit request =>
+    service.get(SubscriptionIdentifier(AppId(rawAppId), ApiContext(rawApiContext), ApiVersion(rawApiVersion))) map {
+      case Some(subscriptionFields) => Ok(Json.toJson(SubscriptionFieldsIdResponse(subscriptionFields.fieldsId)))
+      case None => NOT_FOUND_RESPONSE
+    } recover recovery
   }
-}
 
-object ApiSubscriptionFieldsController extends ApiSubscriptionFieldsController
+  def getSubscriptionFields(rawAppId: UUID, rawApiContext: String, rawApiVersion: String): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
I noticed we have two `getSubscriptionFields` methods that they do exactly the same thing.
We need to verify these are needed. 
If we retrieve the fields only calling one method, we can get rid of the other one (so that we have less testing and less maintenance to do). 
If we need both methods, you could create a private method that is called inside the body of both. The pattern matching used in the `map` is the same code for both. I would write it only once and re-use it

Reply:
Extracted duplicated code as private method.